offix/src/main/java/org/aerogear/offix/ResponseCallback.kt

```diff
@@ -19,4 +19,12 @@ interface ResponseCallback {
      * timeout.
      */
     fun onSchedule(e: ApolloException, mutation: Mutation<Operation.Data, Any, Operation.Variables>)
+
+    /*
+     Called when the library detects error from the server upon Conflict detection.
+     @param serverState: String
+     @param clientState: String
+     @return type:  Mutation<Operation.Data, Any, Operation.Variables>
+     */
+    fun onConflictDetected(serverState: String, clientState: String): Mutation<Operation.Data, Any, Operation.Variables>
```

Comment:
This should be `Mutation<Operation.Data, Any, Operation.Variables>?` so that the user can return null if there was no conflict or if they don't want to handle it.

Reply:
I tried with it, even if we put ? it shows that null can't be a type of not null type Mutation<...> 

The mutation can never be null, that is one issue coming here.

Reply:
Write this in JAVA and then convert it to Kotlin.
Java won't prohibit you from returning null anywhere ;)